site/docs/guides/basic-drawing.md

```diff
@@ -484,11 +522,12 @@ fn pixel(x: i32, y: i32) {
       (local.get $shift)))
 
   ;; Use the first DRAW_COLOR as the pixel color
-  ;; color = *DRAW_COLORS & 0b11;
+  ;; color = *DRAW_COLORS & 0b1111;
   (local.set $color
     (i32.and
       (i32.load16_u (global.get $DRAW_COLORS))
-      (i32.const 3)))
+      (i32.const 15)))
+  ;; FIXME: return if $color is zero, then subtract 1.
```

Comment:
```suggestion
  ;; return if $color is zero, then subtract 1.
  (br_if (i32.eqz (local.get $color)))
  (local.set $color (i32.sub (local.get $color) (i32.const 1)))
```

Reply:
This is the most natural translation of the other code, though you could also subtract 1 ahead of time, and exit the function if the color is < 0